src/software/ai/evaluation/find_passes.cpp

```diff
@@ -0,0 +1,101 @@
+#include "software/ai/evaluation/find_passes.h"
+
+Circle getObstacle(const Robot& robot, double radius)
+{
+    return Circle(robot.position(), radius);
+}
+
+std::vector<Circle> getObstacles(const std::vector<Robot>& robots, double radius)
+{
+    std::vector<Circle> obstacles;
+    for (Robot robot : robots)
+    {
+        obstacles.push_back(getObstacle(robot, radius));
+    }
+    return obstacles;
+}
+
+std::vector<Robot> findOpenFriendlyRobots(const Team& friendly_team,
+                                          const Team& enemy_team, double radius)
+{
+    std::vector<Circle> obstacles = getObstacles(enemy_team.getAllRobots(), radius);
+
+    std::vector<Robot> open_robots;
+    for (Robot friendly : friendly_team.getAllRobots())
+    {
+        if (std::all_of(obstacles.begin(), obstacles.end(),
+                        [friendly](Circle enemy_circle) {
+                            return !contains(enemy_circle, friendly.position());
+                        }))
+        {
+            open_robots.push_back(friendly);
+        }
+    }
+    return open_robots;
+}
+
+std::vector<Pass> findDirectPasses(const Robot& robot, const Team& friendly_team,
+                                   const Team& enemy_team)
+{
+    std::vector<Robot> open_robots =
+        findOpenFriendlyRobots(friendly_team, enemy_team, ROBOT_MAX_RADIUS_METERS * 1.5);
+
+    open_robots.erase(remove(open_robots.begin(), open_robots.end(), robot));
```

Comment:
I remove the robot with the ball from the list of open robots because it would consider itself an "open" robot if there are no nearby enemies so it would then consider a pass to itself a possible direct pass.

Reply:
I suppose that's ok